### PR TITLE
NOISSUE-Update invalid secret error handling

### PR DESCRIPTION
**Pull request by ianmuchyri**
### What does this do?
Updates error handling to handle invalid secret
### Which issue(s) does this PR fix/relate to?
Put here `Resolves #XXX` to auto-close the issue that your PR fixes (if such)
No issue
### List any changes that modify/break current functionality
None
### Have you included tests for your changes?
No
### Did you document any new/modified functionality?
No

### CODE REVIEW

File: users/clients/service.go [hunk 1]
```diff
@@ -315,7 +315,7 @@ func (svc service) UpdateClientSecret(ctx context.Context, token, oldSecret, new
 		return mfclients.Client{}, err
 	}
 	if _, err := svc.IssueToken(ctx, dbClient.Credentials.Identity, oldSecret); err != nil {
-		return mfclients.Client{}, errors.ErrAuthentication
+		return mfclients.Client{}, errors.ErrSecretError
```
Comment (rodneyosodo) [inline users/clients/service.go]: This should return the `IssueToken` error, then we make IssueToken errors more sensible. 

Something like

```suggestion
		return mfclients.Client{}, err
```

of wrap the error

```suggestion
		return mfclients.Client{}, errors.Wrap(errors.ErrAuthentication, err)
```

File: pkg/errors/types.go [hunk 2]
```diff
@@ -42,4 +42,7 @@ var (
 
 	// ErrScanMetadata indicates problem with metadata in db.
 	ErrScanMetadata = New("failed to scan metadata in db")
+
+	//ErrSecretError indicates an error with the entity secret
+	ErrSecretError = New("invalid entity secret")
```
Comment (rodneyosodo) [inline pkg/errors/types.go]: With Gmail as an example

image

Something like

```suggestion
	ErrWrongSecretError = New("wrong secret")
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: pkg/errors/types.go (hunk 2 above)
Comment (rodneyosodo) [reply pkg/errors/types.go]: Also, fix the comment. Add full stop at the end

**Review by dborovcanin — DISMISSED** [review]

File: pkg/errors/types.go [hunk 3]
```diff
@@ -42,4 +42,7 @@ var (
 
 	// ErrScanMetadata indicates problem with metadata in db.
 	ErrScanMetadata = New("failed to scan metadata in db")
+
+	//ErrSecretError indicates an error with the entity secret.
```
Comment (rodneyosodo) [inline pkg/errors/types.go]:
```suggestion
	// ErrWrongSecret indicates a wrong secret was provided.
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: pkg/errors/types.go (hunk 3 above)
Comment (drasko) [reply pkg/errors/types.go]: Please fix this

**Review by drasko — CHANGES_REQUESTED** [review]
Looks good, just a cosmetic fix of comment needed

File: pkg/errors/types.go [hunk 4]
```diff
@@ -42,4 +42,7 @@ var (
...
+	ErrSecretError = New("wrong secret")
```
Comment (rodneyosodo) [inline pkg/errors/types.go]:
```suggestion
	ErrWrongSecret = New("wrong secret")
```

File: pkg/errors/types.go [hunk 5]
```diff
@@ -42,4 +42,7 @@ var (
 
 	// ErrScanMetadata indicates problem with metadata in db.
 	ErrScanMetadata = New("failed to scan metadata in db")
+
+	//ErrWrongSecret indicates a wrong secret was provided.
```
Comment (rodneyosodo) [inline pkg/errors/types.go]:
```suggestion
	// ErrWrongSecret indicates a wrong secret was provided.
```

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

**Review by rodneyosodo — APPROVED** [review]
LGTM

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**